flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/HiveFunctionWrapper.java

```diff
@@ -127,13 +122,8 @@ public Class<UDFType> getUDFClass() throws ClassNotFoundException {
      * @return the UDF deserialized
      */
     private UDFType deserializeUDF() {
-        try {
-            return (UDFType)
-                    SerializationUtilities.deserializeObject(
-                            udfSerializedString, (Class<Serializable>) getUDFClass());
-        } catch (ClassNotFoundException e) {
-            throw new FlinkHiveUDFException(
-                    String.format("Failed to deserialize function %s.", className), e);
-        }
+        return (UDFType)
+                SerializationUtilities.deserializeObject(
+                        udfSerializedString, (Class<Serializable>) getUDFClass());
```

Comment:
Will the deserialization use the classloader of the `functionClz`? Is there any test to cover this code path?

Reply:
Good catch. The deserialization actually will use the current thread's classloader which will cause ClassNotFound issue. Now, I set the classloader for the deserialization manually.